x/axelarnet/keeper/msg_server.go

```diff
@@ -404,6 +404,8 @@ func (s msgServer) RetryIBCTransfer(c context.Context, req *types.RetryIBCTransf
 		return nil, err
 	}
 
+	s.BaseKeeper.DeleteFailedTransfer(ctx, req.ID)
```

Comment:
Instead of deleting, why not decouple creation and enqueue? The problem I see is that a new ID is assigned whenever the transfer is being enqueued.